tests/pure_cpp/smart_holder_poc_test.cpp

```diff
@@ -227,6 +237,25 @@ TEST_CASE("from_unique_ptr+as_shared_ptr", "[S]") {
     REQUIRE(*new_owner == 19);
 }
 
+TEST_CASE("from_unique_ptr_derived+as_unique_ptr_base", "[E]") {
```

Comment:
Could you please make this `"[S]"` here?

`[E]` is for from/to combinations that generate an error, `[S]` for valid combinations.